app/main.py

```diff
@@ -1,3 +1,70 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float | int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | float | int) -> Distance:
+        other = Distance.to_distance(other)
+        return Distance(self.km + other.km)
+
+    def __iadd__(self, other: Distance | float | int) -> Distance:
+        other = Distance.to_distance(other)
+        self.km += other.km
+        return self
+
+    def __mul__(self, other: float | int) -> Distance:
+        Distance.check_is_float(other)
+        return Distance(self.km * other)
+
+    @staticmethod
+    def check_is_float(dist: float) -> None:
+        if not (
+                isinstance(dist, int)
+                or isinstance(dist, float)
+        ):
+            raise TypeError
```

Comment:
The method `check_is_float` checks if the input is either an int or a float, not just a float. Consider renaming this method to something like `check_is_number` to better reflect its functionality.